Source/Core/Core/HW/GCMemcard/GCMemcardDirectory.cpp

```diff
@@ -32,96 +32,65 @@
 const int NO_INDEX = -1;
 static const char* MC_HDR = "MC_SYSTEM_AREA";
 
-int GCMemcardDirectory::LoadGCI(const std::string& file_name, bool current_game_only)
+bool GCMemcardDirectory::LoadGCI(GCIFile gci)
 {
-  File::IOFile gci_file(file_name, "rb");
-  if (gci_file)
+  // check if any already loaded file has the same internal name as the new file
+  for (const GCIFile& alreadyLoadedGci : m_saves)
   {
-    GCIFile gci;
-    gci.m_filename = file_name;
-    gci.m_dirty = false;
-    if (!gci_file.ReadBytes(&(gci.m_gci_header), DENTRY_SIZE))
-    {
-      ERROR_LOG(EXPANSIONINTERFACE, "%s failed to read header", file_name.c_str());
-      return NO_INDEX;
-    }
-
-    std::string gci_filename = gci.m_gci_header.GCI_FileName();
-    for (u16 i = 0; i < m_loaded_saves.size(); ++i)
+    if (gci.m_gci_header.GCI_FileName() == alreadyLoadedGci.m_gci_header.GCI_FileName())
     {
-      if (m_loaded_saves[i] == gci_filename)
-      {
-        PanicAlertT("%s\nwas not loaded because it has the same internal filename as previously "
-                    "loaded save\n%s",
-                    gci.m_filename.c_str(), m_saves[i].m_filename.c_str());
-        return NO_INDEX;
-      }
+      ERROR_LOG(EXPANSIONINTERFACE,
+                "%s\nwas not loaded because it has the same internal filename as previously "
+                "loaded save\n%s",
+                gci.m_filename.c_str(), alreadyLoadedGci.m_filename.c_str());
+      return false;
     }
+  }
 
-    u16 num_blocks = gci.m_gci_header.m_block_count;
-    // largest number of free blocks on a memory card
-    // in reality, there are not likely any valid gci files > 251 blocks
-    if (num_blocks > 2043)
-    {
-      PanicAlertT(
-          "%s\nwas not loaded because it is an invalid GCI.\n Number of blocks claimed to be %u",
-          gci.m_filename.c_str(), num_blocks);
-      return NO_INDEX;
-    }
+  // check if this file has a valid block size
+  // 2043 is the largest number of free blocks on a memory card
+  // in reality, there are not likely any valid gci files > 251 blocks
+  const u16 num_blocks = gci.m_gci_header.m_block_count;
+  if (num_blocks > 2043)
+  {
+    ERROR_LOG(
+        EXPANSIONINTERFACE,
+        "%s\nwas not loaded because it is an invalid GCI.\n Number of blocks claimed to be %u",
```

Comment:
I think this space wasn't supposed to be there? It was in the original line though....
```suggestion
        "%s\nwas not loaded because it is an invalid GCI.\nNumber of blocks claimed to be %u",
```